### ISSUE TITLE
Support for highlighting path between selected nodes

### ISSUE THREAD

**Issue by mestaki** [biocore/empress#491]
Hi team,

I found myself recently needing to investigate the phylogenetic distance  between 2 specific ASVs and thought it might be really useful if Empress could visually highlight the full path between 2 (or more?) user-selected nodes. Apologies if this functionality already exists, couldn't find a way to get what I wanted.

**Comment by fedarko** [comment]
Agreed that this'd be useful! This sort of thing isn't in EMPress yet, but I think it should be doable with some finagling.

I think this might overlap somewhat with the suggestion described in #427 -- I guess the path between any two distinct tips would necessarily pass through their LCA (... at least I think so?), although I suppose "highlighting just the LCA of two tips" and "drawing the path" might be two separate issues to tackle.

**Comment by ElDeveloper** [comment]
This came up again today in the context of being able to investigate how related two COVID-19 genomes are to each other (cc @kwcantrell).

**Comment by rob-knight** [comment]
I agree, would be really useful to have this and to show the tip to tip distance somehow in the UI

**Comment by kwcantrell** [comment]
Agreed, I think this would this feature can help the health team and be generally useful to everyone. I will place this feature on the top of my priority list.

**Comment by mestaki** [comment]
Depending on the use-case 2 different versions of this feature may be helpful:   
1.   Highlighting the full branch of a user defined set of features (with feature ID, taxonomy, etc.) without highlighting the distance to each other
1.  Only highlight the tip-to-tip distances between said selected features.

**Comment by ElDeveloper** [comment]
Great feedback. What do y'all think about having a "select" mode where a user can click on the nodes they want to analyze. As nodes are added the shortest path between nodes is highlighted as well as the nodes that are selected. Simultaneously the "tree statistics tab" should be updated with the average tip-to-tip distance  between selected nodes, and the tip-to-tip distances between selected nodes. The "select" mode should remain active regardless of feature metadata changes, barplot changes, or while searching for nodes.

**Comment by wasade** [comment]
Would this be the node-to-node distance, or would this gather tip-to-tip distances from the tips of the selected clades?

**Comment by fedarko** [comment]
I like @ElDeveloper's idea of a "select" mode. Rather than updating the Tree Statistics tab, I wonder if it might make sense to add a new "Selected Nodes" tab or something similar: this could include a table or something with information about all currently selected nodes (with options to un-select all or some of the currently selected nodes), options to toggle for whether or not the user is selecting one node at a time (probably should remain the default) or multiple nodes at once, etc., as well as statistics about the distances between selected nodes (and options about these stats). I think a fair amount of the work on the shearing panel should be adaptable to this.

Regarding actually computing / showing the distances, there are multiple ways to handle this like @mestaki and @wasade suggested (esp when > 2 nodes are selected). The simplest case might be the most useful (and thus a good place to start): if we set things up so that there's a table of all currently selected nodes, we could adjust the UI to let users select two arbitrary nodes from this table and then show the distance between these nodes (both as a number relative to branch lengths in the tree, and colored on the tree). In any case, having the extra UI space of a new tab should help, since we'll probably end up adding more ways to compute this in the future.

If we can design a general "Selected Nodes" UI in a reusable way, it should open up a lot of future options. For example, the ability to manually assign colors to individual nodes and/or their subtrees directly (#417), or the ability to automatically select multiple features highlighted by Emperor interactions rather than only coloring them in the tree.
